tools/api-coverage/src/main.rs

```diff
@@ -0,0 +1,207 @@
+            self.report_constants();
+        let (mismatch_apis, called_apis, declared_apis, generated_apis) =
+            self.report_calls();
+
+        println!(
+            "Constants:     {} of {} ({:.2}%)",
+            declared_constants,
+            generated_constants,
+            declared_constants as f64 / generated_constants as f64 * 100.0f64
+        );
+
+        println!(
+            "Declared APIs: {} of {} ({:.2}%)",
+            declared_apis,
+            generated_apis,
+            declared_apis as f64 / generated_apis as f64 * 100.0f64
+        );
+
+        println!(
+            "Called APIs:   {} of {} ({:.2}%)",
+            called_apis,
+            generated_apis,
+            called_apis as f64 / generated_apis as f64 * 100.0f64
+        );
+
+        mismatch_constants || mismatch_apis
+    }
+
+    fn report_constants(&self) -> (bool, u64, u64) {
+        let mut constant_names: Vec<String> = Vec::new();
+        for key in self.generated_defs.constants.keys() {
+            constant_names.push(key.clone());
+        }
+        constant_names.sort();
+
+        let mut unwrapped: Vec<String> = Vec::new();
+        let mut declared = 0;
+        let mut generated = 0;
+        let mut mismatch = false;
+
+        for name in constant_names {
+            generated += 1;
+            let constant = self.generated_defs.constants.get(&name).unwrap();
+            if let Some(ignored) = self.ignored_defs.constants.get(&name) {
+                assert_eq!(constant.expr, ignored.expr);
+                continue;
+            }
+
+            let sys_def = self.sys_defs.constants.get(&name);
+            if sys_def.is_none() {
+                unwrapped.push(name);
+            } else if sys_def.unwrap().expr == constant.expr {
+                declared += 1;
+            } else {
+                mismatch = true;
+                println!("CONSTANT MISMATCH: {}", name);
+            }
+        }
+
+        if mismatch {
+            println!();
+        }
+
+        if !unwrapped.is_empty() {
+            println!("Unwrapped Constants:");
+            for name in &unwrapped[..] {
+                println!("  {}", name);
+            }
+            println!();
+        }
+
+        (mismatch, declared, generated)
+    }
+
+    fn report_calls(&self) -> (bool, u64, u64, u64) {
+        let mut signature_names: Vec<String> = Vec::new();
+        for key in self.generated_defs.signatures.keys() {
+            signature_names.push(key.clone());
+        }
+        signature_names.sort();
+
+        let mut uncalled: Vec<String> = Vec::new();
+        let mut unwrapped: Vec<String> = Vec::new();
+        let mut called = 0;
+        let mut declared = 0;
+        let mut generated = 0;
+        let mut mismatch = false;
+
+        for name in signature_names {
+            generated += 1;
+            let sig = self.generated_defs.signatures.get(&name).unwrap();
```

Comment:
Minor but you can avoid the extra hash lookup by doing `self.generated_defs.signatures.iter()` above and collecting into `Vec<(String, String)>` which I am pretty sure will sort on the key first.

Reply:
I see your point, but sig is a `struct syn::Signature` so that'll probably mean sort_on with a lambda, but should be easy enough.

Reply:
Fixed.